sentinel/pkg/checker/dal/app.go

```diff
@@ -0,0 +1,89 @@
+package dal
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"bisonai.com/orakl/sentinel/pkg/alert"
+	"bisonai.com/orakl/sentinel/pkg/request"
+	"bisonai.com/orakl/sentinel/pkg/secrets"
+	"github.com/rs/zerolog/log"
+)
+
+const DefaultDalCheckInterval = 10 * time.Second
+
+type OutgoingSubmissionData struct {
+	Symbol        string   `json:"symbol"`
+	Value         string   `json:"value"`
+	AggregateTime string   `json:"aggregateTime"`
+	Proof         []byte   `json:"proof"`
+	FeedHash      [32]byte `json:"feedHash"`
+	Decimals      string   `json:"decimals"`
+}
+
+func Start() error {
+	interval, err := time.ParseDuration(os.Getenv("DAL_CHECK_INTERVAL"))
+	if err != nil {
+		interval = DefaultDalCheckInterval
+	}
+
+	chain := os.Getenv("CHAIN")
+	if chain == "" {
+		return errors.New("CHAIN not found")
+	}
+
+	key := secrets.GetSecret("DAL_API_KEY")
+	if key == "" {
+		return errors.New("DAL_API_KEY not found")
+	}
+
+	endpoint := fmt.Sprintf("https://dal.%s.orakl.network/api/v1", chain)
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		err := checkDal(endpoint, key)
+		if err != nil {
+			log.Error().Str("Player", "DalChecker").Err(err).Msg("error in checkDal")
+		}
+	}
+	return nil
+}
+
+func checkDal(endpoint string, key string) error {
+	msg := ""
+
+	resp, err := request.Request[[]OutgoingSubmissionData](
+		request.WithEndpoint(endpoint+"/dal/latest-data-feeds/all"),
+		request.WithHeaders(map[string]string{"X-API-Key": key}),
+	)
+	if err != nil {
+		return err
+	}
+
+	for _, data := range resp {
+		rawTimestamp, err := strconv.ParseInt(data.AggregateTime, 10, 64)
+		if err != nil {
+			log.Error().Str("Player", "DalChecker").Err(err).Msg("failed to convert timestamp string to int64")
+			continue
+		}
+
+		timestamp := time.Unix(rawTimestamp, 0)
+		offset := time.Since(timestamp)
```

Comment:
I think this logic will work in an average case. But in an edge case where if there is network delay (or response delay from `DAL`) we'll end up with a false positive. One solution I can think of off the top of my head is to have a new api endpoint on the `DAL` side that returns aggregate data offsets. We can make that endpoint only accessible from sentinel.

Reply:
I think that shouldn't be taken as `false positive`. if it is delayed, it should be somehow investigated and fixed

Reply:
The last time I checked it was having about 200ms network delay 🤔

Reply:
yes that should be the average case. And I'm not sure how realistic is this edge case I mentioned above. If you think it probably wont happen then we can close this discussion 🙏 

The reason I thought this edge case will be a `false positive` is because the error message will say `aggregation has delayed` but in reality it's a network delay. But I think, as you mentioned, it shouldn't be hard to tell if it's aggregation or network delay when we investigate.

Reply:
I see, maybe we can add detailed logic later so that we can tell the difference.

but in a user's perspective, it's probably same, no matter it is network delay or service malfunctioning, not working as expected, so we'll also need to investigate in such cases

Reply:
sounds good, lets merge! 🎉